trunk/examples/LTL/regression/c/vanilla/Bug_ProcedureHandling.c

```diff
@@ -1,54 +1,54 @@
-//#Unsafe
-//@ ltl invariant positive: AP(istemp == 0) || []AP(A != 1);
-
-extern int __VERIFIER_nondet_int() __attribute__ ((__noreturn__));
-
-void StrategyInvalidateBuffer() {}
-void WaitIO() {}
-int RelFileNodeEquals(int a, int b) { return __VERIFIER_nondet_int(); }
-
-int istemp = __VERIFIER_nondet_int();
-int bufHdr_tag_rnode;
-int rnode;
-int A;
-int NLocBuffer;
-int nondet;
-
-void main() {
-	//istemp = __VERIFIER_nondet_int();
-	nondet = __VERIFIER_nondet_int();
-	NLocBuffer = 2;
-	
-	A = 0;
-	if (istemp==1)
-	{
-		 for (int i = 0; i < NLocBuffer; i++)
-		{
-			if (RelFileNodeEquals(bufHdr_tag_rnode, rnode))
-			{
-			}
-		}
-		goto my_exit;
-	}
-	
-	A = 1;
-
-	for (int i = 1; i <= NLocBuffer; i++)
-	{
-
-recheck:
-		if (RelFileNodeEquals(bufHdr_tag_rnode, rnode))
-		{
-			if (nondet)
-			{
-				nondet = __VERIFIER_nondet_int();
-				WaitIO(); 
-				goto recheck;
-			}
-			StrategyInvalidateBuffer(); 
-		}
-	}
-        
- my_exit:
-	while(1) { int yyy;yyy=yyy;}
+//#Unsafe
+//@ ltl invariant positive: <>AP(istemp == 0) || []AP(A != 1);
```

Comment:
Why? The idea here was: either init sets istemp to 0 and then everything is allowed or its some other value which has to be different from 1.

The new property should always be safe b.c. of line 22, making the classification wrong.

Reply:
C does not allow the initialisation of global variables with a non-constant i.e. assigning nondet_int is not valid.
Therefore it is assigned at the first place possible - triggering the finally part of the formula if it is ``0`` (``istemp`` is not written again), or requiring ``A != 1`` wich is correct if the goto in line 31 does work correctly.